shared/actions/push.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import * as Constants from '../constants/push'
+
+// $FlowIssue tell flow about this module
+import * as PushNotifications from 'react-native-push-notification'
+
+import {apiserverPostRpcPromise} from '../constants/types/flow-types'
+
+import {call, put, select} from 'redux-saga/effects'
+import {takeEvery, takeLatest} from 'redux-saga'
+
+import type {SagaGenerator} from '../constants/types/saga'
+import type {TypedState} from '../constants/reducer'
+
+import type {PushNotification, PushNotificationAction, PushPermissionsPromptAction, PushPermissionsRequestAction, PushPermissionsRequestingAction, PushTokenAction, SavePushTokenAction, TokenType, UpdatePushTokenAction} from '../constants/push'
+
+export function permissionsRequest (): PushPermissionsRequestAction {
+  return {type: Constants.permissionsRequest, payload: undefined}
+}
+
+function * permissionsRequestSaga (): SagaGenerator<any, any> {
+  try {
+    yield put({type: Constants.permissionsRequesting, payload: true})
+
+    console.log('Requesting permissions')
+    const permissions = yield call(() => { return PushNotifications.requestPermissions() })
```

Comment:
Why not just do `yield call([PushNotifications, PushNotifications.requestPermissions])`